### DISCUSSION TITLE
Future of the Project!

### DISCUSSION THREAD

**Discussion by nicksawhney** [nicksawhney/bernie-sits#22] [General]
Hey everyone! Thrilled at the attention this site is getting. Let's use this as the forum for discussing how to move forward. I have two objectives here, originally brought up in https://github.com/nicksawhney/bernie-sits/pull/8#issuecomment-766295945
> I'm thinking we maintain a relatively original-looking version of the site, and have another branch where we can go absolutely wild with everything people want to contribute -- styling, different images and overlays (like people have been asking for on twitter), changing image placement, all the rest! That version could become a more general open-source web based meme creation app. Would love anyone's thoughts!

Basically an aesthetically faithful "internet relic" version and an actual functional versoin. 

Some questions to start the discussion are: should the `main` be the original version or the version with contributions? 

Which features should be prioritized?

If you have a change like the recently suggested caching or other backend efficiency improvements, those can go in the original version! 

Let's use this as the forum to figure out what the vibe of this repo will be going forward. If you're experienced maintaining open source, please let me know! I want to do things right.

**Comment by vtaero** [comment]
I like the idea of an original version just as it was when it went crazy. It just placed the image and the webpage interface was super simple.

Then I like a version that has all the little extras. 

I'm trying to make one that I can insert my own pictures into. I have friends I want to place all over the world and I want to make a site where they can do that to others. Obviously I'll need to figure out if there is a cheaper way to get it done without massive cost from google.

**Comment by nicksawhney** [comment]
I think using branches is also the way to go with the two versions, but I am willing to be convinced about using Github Version Releases.

**Comment by keevie** [comment]
Caching seems like it would be pretty straightforward to implement with this https://github.com/reclosedev/requests-cache + redis (since heroku supports that nicely).

**Comment by samarmohan** [comment]
A little bit of styling is necessary for the main site. It is obviously fine right now but if thousands of people are visiting it every day, we need it to look nicer.
Also, you are hosting the site on Heroku but its cold starts are terrible (10-20 seconds), you should try hosting it somewhere else. NOT URGENT

**Reply by nicksawhney** [reply]
The good thing about multiple machines running at once is that updates propogate across machines in such a way that the site technically never goes down. This is, however, an important consideration if a site is deployed on a single hobby dyno.

**Comment by nicksawhney** [comment]
So, I fully intended on adding styling within the first hours after
release. Didn't expect the site to blow up and completely deprioritized
styling amid maintenance and scaling issue, then amid dealing with api
costs. The site is kinda ugly for sure. But the point of the original
branch is to be a relic of the crazy viral site. The ugliness is kind of a
part of the story and character now. I've definitely seen it be a part of
why people found the story of release endearing. For example, from the
wired article linked in the readme:

> In fairness, that the site was thrown together on a whim adds to its charm. It’s not flashy, and it delivers precisely what it promises: pictures of Sanders anywhere in the world that Google’s Street View cars have roamed.

We definitely should have a high design standard for the "go-wild" version
of the site (what should we name these different versions?)

**Comment by Davidgloss** [comment]
Hi! I just signed up to Github to make this one comment. I googled Bernie Sits app and got some sad story about no money for servers. Why don't you tell this sad story to Kickstarter and maybe they can help you raise money to pay for servers and fund further developments?
Btw. Github asked me how much computer programming experience I have. Even though I am 66, I took Fortran in college. Fortran lV in winter trimester of 1972-1973. I learned how to use the punch card machine to write simple programs in Fortran IV. I played the computer game Kirk through a timeshare with the IBM 360 in Princeton. I don't think they use the punch card machines anymore. Is that a lot of programming experience? After all, there just wasn't much programming going on back then.

Please let me know if the app becomes available again.

**Reply by nicksawhney** [reply]
Hey David, so glad you signed up for github! I'm working on a way to get the site back online that I think will develop some useful technology in the process, I'd love your thoughts! See: #25 . 

Your experience is awesome! I'm sure it's amazing to have seen the evolution of computing from such a different stage. How's today's software engineering world different than your expectations back then of how it would be?

**Comment by nicksawhney** [comment]
Here's the idea I just had: https://github.com/nicksawhney/bernie-sits/issues/25

**Comment by itslevir** [comment]
I would suggest using [Microsoft Azure](azure.com). Microsoft Azure tends to cost more if you have less requests, but costs less if you have more requests (that is, compared to other cloud providers like GCP)
